### PR TITLE
refactor: implement sharing NG crud operations

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
Use sharing NG for all share related CRUD operations as well as loading outgoing shares. All that mostly affects the sharing panel in the sidebar.

## Known issues on the server side

- [x] ~Inviting multiple people does not work yet (https://github.com/owncloud/ocis/issues/7962)~ seems that this is not being planed currently, so we need to do this like it was before
- [x] ~Expiration date for links can't be removed (https://github.com/owncloud/ocis/issues/8405)~
- [x] ~Shares can't be created with an expiration date. Only available via update (https://github.com/owncloud/ocis/issues/8404)~
- [x] ~Space members are not being listed because they are missing in the server response (https://github.com/owncloud/ocis/issues/8352) **(blocking)**~
- [ ] OCM shares are not being listed because they are missing in the server response (https://github.com/owncloud/ocis/issues/8564)
- [ ] Available roles include incompatible ones (https://github.com/owncloud/ocis/issues/8331)
- [ ] Getting permissions for a project space is a bit messy currently as Web needs to build the space ID itself (https://github.com/owncloud/ocis/issues/8351) 
- [ ] Tokeninfo endpoint is still missing (https://github.com/owncloud/ocis/issues/8617)
- [x] ~Groups and users with `"` in their name can't be queried (https://github.com/owncloud/ocis/issues/8615)~
- [x] ~Changing the type for an existing link with a password to "internal" does not remove the password, leading to weird behaviours (https://github.com/owncloud/ocis/issues/8619)~
- [ ] Error message for banned passwords is not properly capitalized (https://github.com/owncloud/ocis/issues/8624)
- [x] ~https://github.com/owncloud/ocis/pull/8682~
- [ ] `hidden` property is not editable via sharing NG (https://github.com/owncloud/ocis/issues/8654)

## Open questions

see https://github.com/owncloud/web/issues/10418

## Follow-ups

see https://github.com/owncloud/web/issues/10418

## Related Issue

- refs https://github.com/owncloud/web/issues/10418
- fixes https://github.com/owncloud/web/issues/10669

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

## Open tasks for this PR
- [x] Fix e2e and acceptance tests

### CODE REVIEW

File: tests/e2e/cucumber/features/journeys/kindergarten.feature [hunk 1]
```diff
@@ -41,26 +41,26 @@ Feature: Kindergarten can use web to organize a day
...
+      | groups/Pre-Schools Pirates/meal plan/lorem-big.txt | sales     | group | Can view | file         |
+      | groups/Pre-Schools Pirates/meal plan/lorem-big.txt | Carol     | user  | Can view | file         |
+      | groups/Kindergarten Koalas/meal plan               | sales     | group | Can view | folder       |
+      | groups/Kindergarten Koalas/meal plan               | security  | group | Can edit | folder       |
+      | groups/Kindergarten Koalas/meal plan/lorem.txt     | sales     | group | Can view | file         |
+      | groups/Kindergarten Koalas/meal plan/lorem.txt     | security  | group | Can view | file         |
+      | groups/Teddy Bear Daycare/meal plan                | Brian     | user  | Can edit | folder       |
+      | groups/Teddy Bear Daycare/meal plan                | Carol     | user  | Can edit | folder       |
+      | groups/Teddy Bear Daycare/meal plan/data.zip       | Brian     | user  | Can edit | file         |
+      | groups/Teddy Bear Daycare/meal plan/data.zip       | Carol     | user  | Can edit | file         |
```
Comment (ScharfViktor) [inline tests/e2e/cucumber/features/journeys/kindergarten.feature:56]: here we wanted to check sharing and share updating with custom roles

in the sharing ng we can share using roles and permissions action:

 ```
"roles": [
    "1c996275-f1c9-4e71-abdf-a42f6495e960"
  ]
  "@libre.graph.permissions.actions": [
    "libre.graph/driveItem/permissions/create",
    "libre.graph/driveItem/quota/read"
  ]
```

If we leave the `permissionsAction` feature (sharing with custom permissions) on the web, I wouldn't change this test. if you need a help with e2e just ping me

or we can skip this tests temporaly and change them in separate PR/PRs

Comment (JammingBen) [reply tests/e2e/cucumber/features/journeys/kindergarten.feature:56]: As discussed with @tbsbdr , we won't have custom permissions with sharing NG anymore. That's why I replaced the custom permissions in this test with predefined roles.

File: packages/web-app-files/src/components/SideBar/Shares/Links/DetailsAndEdit.vue [hunk 2]
```diff
@@ -466,61 +469,45 @@ export default defineComponent({
...
-                ...this.link,
-                name
-              }
-            })
-          )
+        onConfirm: (displayName: string) => {
+          const linkShare = this.linkShare
+          linkShare.displayName = displayName
+          return Promise.resolve(this.$emit('updateLink', { linkShare }))
+        }
```
Comment (dschmidt) [inline packages/web-app-files/src/components/SideBar/Shares/Links/DetailsAndEdit.vue:498]: This is completely sync, right?

Semantically it's also really weird to me to wrap the result of $emit in a promise - isn't it empty anyhow?

File: packages/web-app-files/src/components/SideBar/Shares/Links/DetailsAndEdit.vue [hunk 3]
```diff
@@ -535,14 +522,7 @@ export default defineComponent({
...
         onConfirm: (value: string) =>
-          Promise.resolve(
-            this.updateLink({
-              link: {
-                ...this.link,
-                notifyUploadsExtraRecipients: value
-              }
-            })
-          )
+          Promise.resolve(this.$emit('updateLink', { linkShare: { ...this.linkShare } }))
```
Comment (dschmidt) [inline packages/web-app-files/src/components/SideBar/Shares/Links/DetailsAndEdit.vue:525]: same here

File: packages/web-app-files/src/components/SideBar/Shares/SharesPanel.vue [hunk 4]
```diff
@@ -39,23 +38,15 @@ export default defineComponent({
...
       activePanel: inject<String>('activePanel')
     }
   },
   watch: {
     sharesLoading: {
       handler: function (sharesLoading, old) {
-        if (!sharesLoading) {
-          this.loadIncomingParentShare.perform(this.resource)
-        }
         // FIXME: !old can be removed as soon as https://github.com/owncloud/web/issues/7621 has been fixed
```
Comment (dschmidt) [inline packages/web-app-files/src/components/SideBar/Shares/SharesPanel.vue:50]: heads up: this was closed meanwhile, could be resolved (in a follow up)

File: packages/web-app-files/tests/unit/components/SideBar/Shares/Collaborators/__snapshots__/ListItem.spec.ts.snap [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
   <div data-v-ccf14be2="" class="oc-width-1-1 oc-flex oc-flex-middle files-collaborators-collaborator-details">
     <div data-v-ccf14be2="" class="oc-width-2-3 oc-flex oc-flex-middle">
       <div data-v-ccf14be2="">
-        <avatar-image-stub data-v-ccf14be2="" userid="brian" user-name="Brian Murphy" width="36" class="files-collaborators-collaborator-indicator"></avatar-image-stub>
+        <avatar-image-stub data-v-ccf14be2="" userid="3" user-name="einstein" width="36" class="files-collaborators-collaborator-indicator"></avatar-image-stub>
       </div>
       <div data-v-ccf14be2="" class="oc-pl-s oc-text-truncate">
-        <div data-v-ccf14be2="" class="oc-text-truncate"><span data-v-ccf14be2="" aria-hidden="true" class="files-collaborators-collaborator-name">Brian Murphy</span> <span data-v-ccf14be2="" class="oc-invisible-sr">Share receiver name: Brian Murphy (brian@owncloud.com)</span></div>
+        <div data-v-ccf14be2="" class="oc-text-truncate"><span data-v-ccf14be2="" aria-hidden="true" class="files-collaborators-collaborator-name">einstein</span> <span data-v-ccf14be2="" class="oc-invisible-sr">Share receiver name: einstein</span></div>
```
Comment (dschmidt) [inline packages/web-app-files/tests/unit/components/SideBar/Shares/Collaborators/__snapshots__/ListItem.spec.ts.snap:11]: Is it desired that the full name plus email was replaced with just the username here?

File: packages/web-app-files/tests/unit/components/SideBar/Shares/FileShares.spec.ts [hunk 6]
```diff
@@ -79,20 +73,16 @@ describe('FileShares', () => {
...
     it('reacts on delete events', async () => {
-      const spyOnCollaboratorDeleteTrigger = vi
-        .spyOn(FileShares.methods, '$_ocCollaborators_deleteShare_trigger')
-        .mockImplementation(() => undefined)
       const { wrapper } = getWrapper({ collaborators })
-      wrapper
-        .findComponent<typeof CollaboratorListItem>('collaborator-list-item-stub')
-        .vm.$emit('onDelete')
+      const { dispatchModal } = useModals()
+      ;(wrapper.findComponent<any>('collaborator-list-item-stub').vm as any).$emit('onDelete')
```
Comment (dschmidt) [inline packages/web-app-files/tests/unit/components/SideBar/Shares/FileShares.spec.ts:78]: meh, can't we use proper types with stubs? :-\

File: tests/e2e/cucumber/features/shares/share.feature [hunk 7]
```diff
@@ -22,10 +22,10 @@ Feature: share
...
       | lorem-big.txt | folder_to_customShared |
     When "Alice" shares the following resource using the sidebar panel
-      | resource               | recipient | type | role                                  | resourceType |
-      | folder_to_shared       | Brian     | user | Can edit                              | folder       |
-      | shared_folder          | Brian     | user | Can edit                              | folder       |
-      | folder_to_customShared | Brian     | user | custom_permissions:read,create,delete | folder       |
+      | resource               | recipient | type | role       | resourceType |
+      | folder_to_shared       | Brian     | user | Can edit   | folder       |
+      | shared_folder          | Brian     | user | Can edit   | folder       |
+      | folder_to_customShared | Brian     | user | Can edit   | folder       |
```
Comment (dschmidt) [inline tests/e2e/cucumber/features/shares/share.feature:28]: Mh does this still have any value now? Maybe it should be removed (in follow up)

**Review by dschmidt — APPROVED** [review]
crazy! 🤩

**Comment by JammingBen** [comment]
Thx 😍 As discussed, merging it now - will do the little things in a follow-up after.

File: packages/web-app-files/src/components/SideBar/Shares/SharesPanel.vue (hunk 4 above)
Comment (JammingBen) [reply packages/web-app-files/src/components/SideBar/Shares/SharesPanel.vue:50]: Added to the checks in https://github.com/owncloud/web/issues/10418.

File: packages/web-app-files/src/components/SideBar/Shares/Links/DetailsAndEdit.vue (hunk 3 above)
Comment (JammingBen) [reply packages/web-app-files/src/components/SideBar/Shares/Links/DetailsAndEdit.vue:525]: https://github.com/owncloud/web/pull/10677

File: packages/web-app-files/tests/unit/components/SideBar/Shares/FileShares.spec.ts (hunk 6 above)
Comment (JammingBen) [reply packages/web-app-files/tests/unit/components/SideBar/Shares/FileShares.spec.ts:78]: https://github.com/owncloud/web/pull/10677

File: tests/e2e/cucumber/features/shares/share.feature (hunk 7 above)
Comment (JammingBen) [reply tests/e2e/cucumber/features/shares/share.feature:28]: https://github.com/owncloud/web/pull/10677

**Pull request merged**